### PR TITLE
override_subnet_parameter_from_host_no_impact

### PR DESCRIPTION
**Pull request by vikask18**
Tested with Satellite version:
- 6.5.z
- 6.6.0

Test Results:
-----------------
venvrobottelo)$ pytest tests/foreman/api/test_subnet.py::ParameterizedSubnetTestCase::test_positive_subnet_parameters_override_impact_on_subnet -v
======================= test session starts ===============================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.8.0, pluggy-0.11.0 -- /home/vkaushik/RobotteloFramework/robottelo/venvrobottelo65/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/vkaushik/RobotteloFramework/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0, forked-1.0.2
collecting ... 2019-06-06 15:44:56 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                             

tests/foreman/api/test_subnet.py::ParameterizedSubnetTestCase::test_positive_subnet_parameters_override_impact_on_subnet PASSED                        [100%]

======================== 1 passed in 5.98 seconds ==========================

### CODE REVIEW

File: tests/foreman/api/test_subnet.py [hunk 1]
```diff
@@ -278,6 +277,58 @@ def test_positive_subnet_parameters_override_impact_on_subnet(self):
         CaseLevel: System
         """
 
+        # Create subnet with valid parameters
+        parameters = [
+            {'name': gen_string('alpha'), 'value': gen_string('alpha')}
+        ]
```
Comment (vijay8451) [inline tests/foreman/api/test_subnet.py]:
```suggestion
     parameter = [{'name': gen_string('alpha'), 'value': gen_string('alpha')}]
```
in single line it should fit .

File: tests/foreman/api/test_subnet.py [hunk 2]
```diff
@@ -278,6 +277,58 @@ def test_positive_subnet_parameters_override_impact_on_subnet(self):
...
+            subnet=org_subnet,
+        ).create()
+        self.assertEqual(host.subnet.read().name, org_subnet.name)
+
+        # Change host parameter value and check that parameter override
+        # is successful without affecting subnet parameter
+        name = org_subnet.subnet_parameters_attributes[0]['name']
+        parameters = [{
+            'name': name, 'value': gen_string('alpha')
+        }]
```
Comment (vijay8451) [inline tests/foreman/api/test_subnet.py]: as `name` variable is only using here by `parameter` hence can use in below way
```
parameter = [{'name': '{}'.format(org_subnet.subnet_parameters_attributes[0]['name']), 
    'value': gen_string('alpha')}]
```

File: tests/foreman/api/test_subnet.py [hunk 3]
```diff
@@ -278,6 +277,58 @@ def test_positive_subnet_parameters_override_impact_on_subnet(self):
...
+            location=loc,
+            organization=org,
+            subnet=org_subnet,
+        ).create()
+        self.assertEqual(host.subnet.read().name, org_subnet.name)
+
+        # Change host parameter value and check that parameter override
+        # is successful without affecting subnet parameter
+        name = org_subnet.subnet_parameters_attributes[0]['name']
+        parameters = [{
```
Comment (vijay8451) [inline tests/foreman/api/test_subnet.py]:
```suggestion
        new_parameter = [{
```
rename above i.e. `new_parameter` to make the difference with `parameters`.

File: tests/foreman/api/test_subnet.py [hunk 4]
```diff
@@ -278,6 +277,58 @@ def test_positive_subnet_parameters_override_impact_on_subnet(self):
         CaseLevel: System
         """
 
+        # Create subnet with valid parameters
+        parameters = [
```
Comment (vijay8451) [inline tests/foreman/api/test_subnet.py]:
```suggestion
        parameter = [
```
As only one parameter is create hence more appropriate name would be `parameter` ..

File: tests/foreman/api/test_subnet.py [hunk 5]
```diff
@@ -278,6 +277,58 @@ def test_positive_subnet_parameters_override_impact_on_subnet(self):
...
+        # Create host with above subnet
+        host = entities.Host(
+            location=loc,
+            organization=org,
+            subnet=org_subnet,
+        ).create()
+        self.assertEqual(host.subnet.read().name, org_subnet.name)
+
+        # Change host parameter value and check that parameter override
+        # is successful without affecting subnet parameter
```
Comment (vijay8451) [inline tests/foreman/api/test_subnet.py]: Please remove above doc string as same is already mentioned test doc string steps and expected results.

File: tests/foreman/api/test_subnet.py (hunk 1 above)
Comment (vikask18) [reply tests/foreman/api/test_subnet.py]: Done.

File: tests/foreman/api/test_subnet.py (hunk 2 above)
Comment (vikask18) [reply tests/foreman/api/test_subnet.py]: Done.

File: tests/foreman/api/test_subnet.py (hunk 3 above)
Comment (vikask18) [reply tests/foreman/api/test_subnet.py]: Done.

File: tests/foreman/api/test_subnet.py (hunk 5 above)
Comment (vikask18) [reply tests/foreman/api/test_subnet.py]: Done.

File: tests/foreman/api/test_subnet.py [hunk 6]
```diff
@@ -278,6 +277,52 @@ def test_positive_subnet_parameters_override_impact_on_subnet(self):
...
+        )
+        # Create host with above subnet
+        host = entities.Host(
+            location=loc,
+            organization=org,
+            subnet=org_subnet,
+        ).create()
+        self.assertEqual(host.subnet.read().name, org_subnet.name)
+        new_parameter = [{
+            'name': '{}'.format(org_subnet.subnet_parameters_attributes[0]['name']),
```
Comment (oshtaier) [inline tests/foreman/api/test_subnet.py]: `'{}'.format(org_subnet.subnet_parameters_attributes[0]['name'])`?
That operation does nothing, so please use variable directly

File: tests/foreman/api/test_subnet.py [hunk 7]
```diff
@@ -278,6 +277,52 @@ def test_positive_subnet_parameters_override_impact_on_subnet(self):
...
+            'value': gen_string('alpha')
+        }]
+        host.host_parameters_attributes = new_parameter
+        host = host.update(['host_parameters_attributes'])
+        self.assertEqual(
```
Comment (oshtaier) [inline tests/foreman/api/test_subnet.py]: That validation is redundant as `new_parameter[0]['name']` is equal to `org_subnet.subnet_parameters_attributes[0]['name']`, so remove it

**Comment by oshtaier** [comment]
Use separate commit for addressing comments, so reviewer can see diff, not go through entire PR once more

File: tests/foreman/api/test_subnet.py (hunk 6 above)
Comment (vikask18) [reply tests/foreman/api/test_subnet.py]: Done.

File: tests/foreman/api/test_subnet.py (hunk 7 above)
Comment (vikask18) [reply tests/foreman/api/test_subnet.py]: Done.

**Comment by vikask18** [comment]
Test Result:
----------------
(venvrobottelo)$ pytest tests/foreman/api/test_subnet.py::ParameterizedSubnetTestCase::test_positive_subnet_parameters_override_impact_on_subnet -v
======================== test session starts ============================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.8.0, pluggy-0.11.0 -- /home/vkaushik/RobotteloFramework/robottelo/venvrobottelo65/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/vkaushik/RobotteloFramework/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0, forked-1.0.2
collecting ... 2019-06-07 18:25:29 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                             

tests/foreman/api/test_subnet.py::ParameterizedSubnetTestCase::test_positive_subnet_parameters_override_impact_on_subnet PASSED                        [100%]

======================= 1 passed in 6.72 seconds =========================
(venvrobottelo)$

**Comment by vikask18** [comment]
Test Result:
----------------

(venvrobottelo)$ pytest tests/foreman/api/test_subnet.py::ParameterizedSubnetTestCase::test_positive_subnet_parameters_override_impact_on_subnet -v
======================= test session starts ===========================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.8.0, pluggy-0.11.0 -- /home/vkaushik/RobotteloFramework/robottelo/venvrobottelo65/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/vkaushik/RobotteloFramework/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0, forked-1.0.2
collecting ... 2019-06-07 19:45:17 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                             

tests/foreman/api/test_subnet.py::ParameterizedSubnetTestCase::test_positive_subnet_parameters_override_impact_on_subnet PASSED                        [100%]

======================= 1 passed in 5.31 seconds ======================
(venvrobottelo)]$

**Pull request merged**